onnx/defs/tensor/defs.cc

```diff
@@ -1085,7 +1085,21 @@ ONNX_OPERATOR_SET_SCHEMA(
             "T",
             OpSchema::all_tensor_types(),
             "Constrain input and output types to all tensor types.")
-        .SetDoc(Upsample_ver7_doc));
+        .SetDoc(Upsample_ver7_doc)
+        .TypeAndShapeInferenceFunction([](InferenceContext& ctx) {
+          propagateShapeAndTypeFromFirstInput(ctx);
+          auto& input_shape = getInputShape(ctx, 0);
```

Comment:
Please call `hasNInputShapes(ctx, 1)` to make sure we have the shape information of the first input.